invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -395,8 +395,10 @@ public void createInvoice(final InvoiceModelDao invoice,
                                 billingEventSqlDao.create(new InvoiceBillingEventModelDao(invoiceModelDao.getId(), BillingEventSerializer.serialize(billingEvents), context.getCreatedDate()), context);
                             }
                             createdInvoiceIds.add(invoiceModelDao.getId());
-                        } else if (invoiceOnDisk.getStatus() == InvoiceStatus.DRAFT && invoiceModelDao.getStatus() == InvoiceStatus.COMMITTED) {
-                            invoiceSqlDao.updateStatus(invoiceModelDao.getId().toString(), InvoiceStatus.COMMITTED.toString(), context);
+                        } else if (invoiceOnDisk.getStatus() != invoiceModelDao.getStatus() || /* Update if status is different */
+                                   (invoiceModelDao.getTargetDate() != null && /* Update if target date is specified and prev was either null or different */
+                                    (invoiceOnDisk.getTargetDate() == null || invoiceOnDisk.getTargetDate().compareTo(invoiceModelDao.getTargetDate()) != 0))) {
```

Comment:
I'm wondering if you need to explicitly check that `invoiceOnDisk.getTargetDate().compareTo(invoiceModelDao.getTargetDate()) < 0`, to make sure we never go back in time.

Could you trigger a target date bill run in the past in your test, before committing the final invoice, and verify the resulting target date?